c_include/fine.hpp

```diff
@@ -1200,6 +1209,26 @@ class Registration {
     return {};
   }
 
+  // Registers a load callback.
+  static LoadCallback register_load(LoadCallback callback) {
+    if (erl_nif_load_callback) {
+      throw std::logic_error("load callback already registered");
+    }
+
+    Registration::erl_nif_load_callback = callback;
+    return callback;
+  }
+
+  // Registers an unload callback.
+  static UnloadCallback register_unload(UnloadCallback callback) {
+    if (erl_nif_unload_callback) {
+      throw std::logic_error("unload callback already registered");
+    }
+
+    Registration::erl_nif_unload_callback = callback;
+    return callback;
+  }
```

Comment:
Actually I would do the same as other registrations and return an empty Registration object:

```suggestion
  // Registers a load callback.
  static Registration register_load(LoadCallback callback) {
    if (erl_nif_load_callback) {
      throw std::logic_error("load callback already registered");
    }

    Registration::erl_nif_load_callback = callback;
    return {};
  }

  // Registers an unload callback.
  static Registration register_unload(UnloadCallback callback) {
    if (erl_nif_unload_callback) {
      throw std::logic_error("unload callback already registered");
    }

    Registration::erl_nif_unload_callback = callback;
    return {};
  }
```